src/picongpu/include/plugins/PluginController.hpp

```diff
@@ -98,136 +98,112 @@ class PluginController : public ILightweightPlugin
 
     std::list<ISimulationPlugin*> plugins;
 
-#if (ENABLE_ELECTRONS == 1)
-#if(PIC_ENABLE_PNG==1)
-    typedef Visualisation<PIC_Electrons, PngCreator> ElectronsPngBuilder;
-    typedef PngPlugin<ElectronsPngBuilder > PngImageElectrons;
-#endif
-
-#if(ENABLE_HDF5 == 1)
-    /* speciesParticleShape::ParticleShape::ChargeAssignment */
-    typedef PhaseSpaceMulti<particles::shapes::Counter::ChargeAssignment, PIC_Electrons> PhaseSpaceElectrons;
-#endif
-#if(SIMDIM==DIM3)
+    template<typename T_Type>
+    struct PushBack
+    {
 
-    typedef SliceFieldPrinterMulti<FieldE> SliceFieldEPrinter;
-    typedef SliceFieldPrinterMulti<FieldB> SliceFieldBPrinter;
-    typedef SliceFieldPrinterMulti<FieldJ> SliceFieldJPrinter;
-#endif
+        template<typename T>
+        void operator()(T& list)
+        {
+            list.push_back(new T_Type());
+        }
+    };
 
-    typedef LiveViewPlugin<PIC_Electrons > LiveImageElectrons;
-    typedef CountParticles<PIC_Electrons> ElectronCounter;
-    typedef EnergyParticles<PIC_Electrons> EnergyElectrons;
-    typedef PositionsParticles<PIC_Electrons> PositionElectrons;
-    typedef BinEnergyParticles<PIC_Electrons> BinEnergyElectrons;
-#if(ENABLE_RADIATION == 1)
-    typedef Radiation<PIC_Electrons> RadiationElectrons;
-#endif
-#endif
+    /** apply vector component one to component zero
```

Comment:
- if it is `vector<dataType,plugin>` than you apply el. 0 in el. 1 :)
-> `apply the 1st vector component to the 2nd`